nativelib/src/main/scala/scala/scalanative/runtime/Shutdown.scala

```diff
@@ -0,0 +1,35 @@
+package scala.scalanative.runtime
+
+import scala.collection.mutable
+import scala.scalanative.native._
+
+private[runtime] object Shutdown {
+  def runHooks(): Unit = {
+    hooks.toSeq.sortBy(_._1).foreach {
+      case (_, runnables) =>
+        runnables.foreach(runnable =>
+          try {
+            runnable.run()
+          } catch {
+            case e: Exception => // Maybe add a system propery that adds logging of exceptions?
+        })
+    }
+  }
+  def addHook(id: Int, runnable: Runnable) =
+    hooks.synchronized(hooks.get(id) match {
+      case Some(runnables) => runnables += runnable
+      case None =>
+        val runnables = mutable.Set(runnable)
+        hooks += id -> runnables
+    })
+  private[this] val hooks: mutable.Map[Int, mutable.Set[Runnable]] =
+    mutable.Map.empty
```

Comment:
I don't think the `id` is necessary here. Lets just make it a `mutable.Seq[Runnable]`. Every time you add a hook it adds one more hook to run, in the order they were added.

Reply:
The `id` variable name was bad. It should have been priority. On the jvm, shutdown hooks have a priority that determine the order in which they run. For example, on the jvm the delete on exit hooks are always the last to run. The hooks added via `Runtime.addShutdownHook(Thread)` should run first. I didn't add that api because the current Thread implementation has a private constructor so there is no way for a user to add it at the moment. Leaving in the priority field, it would be trivial in the future to add `Runtime.addShutdownHook`. If that really isn't a concern, then we can go ahead and switch to a Seq now.